paymentapi/src/main/java/fi/hel/verkkokauppa/payment/paytrail/validation/PaytrailRefundReturnValidator.java

```diff
@@ -0,0 +1,86 @@
+package fi.hel.verkkokauppa.payment.paytrail.validation;
+
+import fi.hel.verkkokauppa.common.error.CommonApiException;
+import fi.hel.verkkokauppa.common.error.Error;
+import fi.hel.verkkokauppa.common.rest.CommonServiceConfigurationClient;
+import fi.hel.verkkokauppa.common.util.StringUtils;
+import fi.hel.verkkokauppa.payment.api.data.refund.RefundReturnDto;
+import fi.hel.verkkokauppa.payment.model.refund.RefundPayment;
+import fi.hel.verkkokauppa.payment.repository.refund.RefundPaymentRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.helsinki.paytrail.constants.RefundStatus;
+import org.helsinki.paytrail.service.PaytrailSignatureService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+import java.util.TreeMap;
+
+
+@Service
+@Slf4j
+public class PaytrailRefundReturnValidator {
+
+    @Autowired
+    private RefundPaymentRepository refundPaymentRepository;
+
+    @Autowired
+    private CommonServiceConfigurationClient commonServiceConfigurationClient;
+
+    public boolean validatePaytrailChecksum(Map<String,String> checkoutParams, String merchantId, String signature, String refundId) {
+        RefundPayment refund = refundPaymentRepository.findById(refundId).orElseThrow(() -> new CommonApiException(
```

Comment:
This should not be inside validatePaytrailChecksum function, you could move this to one layer up, to the paytrailRefundpaymentService, function name could be like ´getRefundPaymentWithRefundId´ and just pass the refund id to this function as it is now,

Reply:
Good note, even better was finding that we already had such a method in paytrailRefundpaymentService that we could use as is, so I'll remove this line of code from the validation. Since shopInShop will be implemented later for refunds, we do not have a need for RefundPayment object in this method yet, so I'll leave it off for now until we actually need it.